### PR TITLE
Color maths 4

### LINKED ISSUE #12202: Add color animation support
There are four sections to this issue:

- [ ] Implement `Animatable` for all of our color types.
- [x] Implement `Point` (in the sense of `bevy_math` for all of our color types.
- [x] Figure out the correct way to handle color addition/subtraction and scalar multiplication and division.
- [ ] Add examples demonstrating color animation

Our options for handling alpha during color addition and subtraction from https://github.com/bevyengine/bevy/pull/12163#discussion_r1507773413:

1. Average: leads to fairly surprising behavior, non-associative.
2. Use the left-hand alpha: fairly surprising, non-commutative.
3. Set the alpha to 1: very surprising, associative and commutative.
4. Don't support adding colors with alpha: panic, remove the implementation, or create a set of alpha-free colors.
5. [Just add and subtract alpha like RGB](https://github.com/bevyengine/bevy/pull/12163#discussion_r1507793882): simple, but leads to values outside of 0-1.0 very easily.
6. [Apply alpha blending](https://github.com/bevyengine/bevy/pull/12163#discussion_r1507810743): a little more expensive, probably well-behaved.

### PR DESCRIPTION
**Pull request by solis-lumine-vorago**
# Objective

- Fixes #12202 

## Solution

- This PR implements componentwise (including alpha) addition, subtraction and scalar multiplication/division for some color types.
- The mentioned color types are `Laba`, `Oklaba`, `LinearRgba` and `Xyza` as all of them are either physically or perceptually linear as mentioned by @alice-i-cecile in the issue.

---

## Changelog

- Scalar mul/div for `LinearRgba` may modify alpha now.

## Migration Guide

- Users of scalar mul/div for `LinearRgba` need to be aware of the change and maybe use the `.clamp()` methods or manually set the `alpha` channel.

### CODE REVIEW

File: crates/bevy_color/src/lib.rs
Comment (NthTensor) [inline crates/bevy_color/src/lib.rs:72]: We probably should link to `Mix` here.

File: crates/bevy_color/src/lib.rs
Comment (NthTensor) [inline crates/bevy_color/src/lib.rs:170]: Should these clamp by default?

Comment (solis-lumine-vorago) [reply crates/bevy_color/src/lib.rs:170]: I think this would cause a lot of the splines code to work incorrectly with colors.

**Comment by NthTensor** [comment]
Looking great!

I'll just repeat what I said on discord here as well: Eventually I want to have better support for non-linear alpha blending modes (perhaps with some sort of `Blend<LinearRgba, BlendMode>` or perhaps not). I also want to make converting vectors into colors a bit easier.  Ditto with math on cylindrical color spaces. None of that needs to block this PR.

Mostly what I see remaining is bike-shedding about the specific implementation of the color algebra. The operations don't need to be perfect but they do need to be well defined.

**Review by NthTensor — APPROVED** [review]

File: crates/bevy_color/src/lib.rs
Comment (NthTensor) [reply crates/bevy_color/src/lib.rs:170]: Good point.

Comment (IQuick143) [reply crates/bevy_color/src/lib.rs:170]: Bounding indeed breaks many things in splines.

**Comment by IQuick143** [comment]
> Rereading [#12163 (comment)](https://github.com/bevyengine/bevy/pull/12163#discussion_r1507807808) it looks like we might want to exclude alpha from scalar multiplication/division.

That would break splines, as it breaks distributivity over addition.

**Comment by NthTensor** [comment]
After looking at some other color crates, I am less concerned about the alpha component. We've got enough approvals for merge, and this isn't marked controversial so I am adding the tag.

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**
